JujuHooks/JujuHooks.psm1

```diff
@@ -0,0 +1,1225 @@
+        [Alias("RelId")]
+        [string]$RelationId=$null
+    )
+    PROCESS {
+        $relationData = @()
+
+        $relatedUnits = Get-JujuRelatedUnits -RelationId $RelationId
+        foreach ($i in $relatedUnits) {
+            $unitData = Get-JujuRelationForUnit -Unit $i -RelationId $RelationId
+            $relationData += $unitData
+        }
+        return $relationData
+    }
+}
+
+function Get-JujuRelationsOfType {
+    <#
+    .SYNOPSIS
+     Get relations of a specific type
+    .PARAMETER Relation
+     The name of the relation to get information for
+    #>
+    [CmdletBinding()]
+    Param(
+        [Alias("RelType")]
+        [string]$Relation=$null
+    )
+    PROCESS {
+        $relationData = @()
+        if (!$Relation) {
+            $Relation = Get-JujuRelationType
+            if(!$Relation) {
+                Throw "Could not find relation"
+            }
+        }
+        $relationIds = Get-JujuRelationIds -Relation $Relation
+        foreach ($i in $relationIds) {
+            $relForId = Get-JujuRelationForId $i
+            foreach ($j in $relForId) {
+                $relationData += $j
+            }
+        }
+        return $relationData
+    }
+}
+
+function Confirm-JujuRelationCreated {
+    <#
+    .SYNOPSIS
+     Determine whether or not the relation has been made.
+    .PARAMETER Relation
+     Relation name to check
+    #>
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$Relation,
+        [Obsolete("This parameter is obsolete. It is not required any more.")]
+        [string]$Keys='private-address'
+    )
+    PROCESS {
+        $ids = Get-JujuRelationIds -Relation $Relation
+        if($ids){
+            return $true
+        }
+        return $false
+    }
+}
+
+function Get-JujuUnit {
+    <#
+    .SYNOPSIS
+     returns information about the local unit. It accepts a single argument, which must
+     be private-address or public-address. It is not affected by context.
+
+     See:
+
+     https://jujucharms.com/docs/stable/authors-hook-environment#unit-get
+
+    .PARAMETER Attribute
+     This parameter must be either private-address or public-address
+    .NOTES
+     On some providers both the private-address and public-address may be hostnames instead of IP addresses,
+     so please take that into account when using this function. If you need the actual IP address instead of a
+     hostname, you will have to validate the result, or use Get-JujuUnitPrivateIP as a helper function.
+    #>
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$true)]
+        [ValidateSet("private-address", "public-address")]
+        [Alias("Attr")]
+        [string]$Attribute
+    )
+    PROCESS {
+        $cmd = @("unit-get.exe", "--format=json", $Attribute)
+        return (Invoke-JujuCommand $cmd | ConvertFrom-Json)
+    }
+}
+
+function Confirm-IP {
+    <#
+    .SYNOPSIS
+     Check if the parameter passed as a string, is a valid IPv4 address
```

Comment:
It will return `$true` even for valid IPv6 addresses. Need to complete description.